flink-table-store-common/src/test/java/org/apache/flink/table/store/data/serializer/SerializerTestBase.java

```diff
@@ -0,0 +1,351 @@
+
+    protected abstract boolean deepEquals(T t1, T t2);
+
+    protected abstract T[] getTestData();
+
+    // --------------------------------------------------------------------------------------------
+
+    @Test
+    protected void testCopy() {
+        try {
+            Serializer<T> serializer = getSerializer();
+            T[] testData = getData();
+
+            for (T datum : testData) {
+                T copy = serializer.copy(datum);
+                checkToString(copy);
+                deepEquals("Copied element is not equal to the original element.", datum, copy);
+            }
+        } catch (Exception e) {
+            System.err.println(e.getMessage());
+            e.printStackTrace();
+            fail("Exception in test: " + e.getMessage());
+        }
+    }
+
+    @Test
+    void testSerializeIndividually() {
+        try {
+            Serializer<T> serializer = getSerializer();
+            T[] testData = getData();
+
+            for (T value : testData) {
+                TestOutputView out = new TestOutputView();
+                serializer.serialize(value, out);
+                TestInputView in = out.getInputView();
+
+                assertTrue("No data available during deserialization.", in.available() > 0);
+
+                T deserialized = serializer.deserialize(in);
+                checkToString(deserialized);
+
+                deepEquals("Deserialized value if wrong.", value, deserialized);
+
+                assertTrue("Trailing data available after deserialization.", in.available() == 0);
+            }
+        } catch (Exception e) {
+            System.err.println(e.getMessage());
+            e.printStackTrace();
+            fail("Exception in test: " + e.getMessage());
+        }
+    }
+
+    @Test
+    void testSerializeAsSequenceNoReuse() {
+        try {
+            Serializer<T> serializer = getSerializer();
+            T[] testData = getData();
+
+            TestOutputView out = new TestOutputView();
+            for (T value : testData) {
+                serializer.serialize(value, out);
+            }
+
+            TestInputView in = out.getInputView();
+
+            int num = 0;
+            while (in.available() > 0) {
+                T deserialized = serializer.deserialize(in);
+                checkToString(deserialized);
+
+                deepEquals("Deserialized value if wrong.", testData[num], deserialized);
+                num++;
+            }
+
+            assertEquals("Wrong number of elements deserialized.", testData.length, num);
+        } catch (Exception e) {
+            System.err.println(e.getMessage());
+            e.printStackTrace();
+            fail("Exception in test: " + e.getMessage());
+        }
+    }
+
+    @Test
+    void testSerializabilityAndEquals() {
+        try {
+            Serializer<T> ser1 = getSerializer();
+            Serializer<T> ser2;
+            try {
+                ser2 = SerializationUtils.clone(ser1);
```

Comment:
We should use `org.apache.flink.table.store.utils.InstantiationUtil`.

Reply:
> We should use `org.apache.flink.table.store.utils.InstantiationUtil`.
I believe this could fix the pre-commit build error.